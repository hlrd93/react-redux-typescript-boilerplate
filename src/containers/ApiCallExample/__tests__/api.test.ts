const testData = [1,2,3];

const axiosGetMock = jest.fn(() => ({
  then: (f) => f({ data: testData })
}));

jest.mock('axios', () => ({
  default: { get: axiosGetMock },
}));

import { fetchData } from '../api';

describe('ApiCallExample API', () => {

  it('calls and handles fetchData result correctly', () => {
    const result = fetchData();
    expect(axiosGetMock.mock.calls[0]).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

});
