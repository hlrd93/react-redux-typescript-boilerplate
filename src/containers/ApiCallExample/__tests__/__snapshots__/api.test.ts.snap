exports[`ApiCallExample API calls and handles fetchData result correctly 1`] = `
Array [
  "/api/get-data",
]
`;

exports[`ApiCallExample API calls and handles fetchData result correctly 2`] = `
Array [
  2,
]
`;
